[Delivers #149928779] Containerize Apps

jonajonlee opened the pull request:
Dependencies
- #16 
- #18

Commit:
Merge branch 'bug/149901790-BucketNotCreated' into feature/149928779-Dockerize

Commit:
Merge branch 'feature/149494619-WebClient-ErrorMessages' into feature/149928779-Dockerize

Commit:
Surface error name and message in listUrls route

Commit:
Travis CI: Make specific docker-compose file only for CI

Commit:
Remove STAGE environment variable from package.json where not needed

Commit:
Dockerize deployment of the s3-photos app

Commit:
[Finishes #149928779] Containerize Apps

Commit:
Add unit tests for new cases

Commit:
Change footer to include Cloud Assessments

Commit:
Auto-Generate identifier to be used in S3 Bucket Name

sreimer15 reviewed (approved):
Looks good to me,
one thing that definitely isn't worth changing but wanted to make you aware of

if you have upload good pngs and then upload a bad gif the error message that is rendered hides the successful images.

This is probably because a page is being used as the error page instead of an error div that is being shown and hidden.

Commit:
WebClient - assertDynamoTable function that will ensure that a table exists in the user’s dynamodb table

Commit:
WebClient - Middleware getOrCreateS3BucketId: get or create the S3BucketId for the web-client and set in app.locals

Commit:
- WebClient: run assertDynamoDbTable when the web-client is deployed
- WebClient: run getOrCreateS3BucketId on every route call in case dynamodb is not finished initializing
- WebClient: replace usage of req.deps in favor of documented app.locals

Commit:
- Add eslint config to each app for the linter to work for the app by itself
- Change Travis CI to build and test each app individually
- Removed docker-compose deployment method since it is now defunct given aws role is provided by the EC2 container

Commit:
Merge branch 'staging' of https://github.com/CloudAssessments/s3photoapp into feature/149928779-Dockerize
